Mathlib/Data/Real/Basic.lean

```diff
@@ -607,3 +607,11 @@ def IsPowMul {R : Type*} [Pow R ℕ] (f : R → ℝ) :=
 def RingHom.IsBoundedWrt {α : Type*} [Ring α] {β : Type*} [Ring β] (nα : α → ℝ) (nβ : β → ℝ)
     (f : α →+* β) : Prop :=
   ∃ C : ℝ, 0 < C ∧ ∀ x : α, nβ (f x) ≤ C * nα x
+
+/-- If `x` is multiplicative with respect to `f`, then so is any `x^n`. -/
+theorem pow_isMul_of_isMul {R : Type _} [CommRing R] (f : R → ℝ) {x : R}
+    (hx : ∀ y : R, f (x * y) = f x * f y) : ∀ (n : ℕ) (y : R), f (x ^ n * y) = f x ^ n * f y := by
+  intro n
+  induction n with
+  | zero => intro y; rw [pow_zero, pow_zero, one_mul, one_mul]
+  | succ n hn => intro y; rw [pow_succ', pow_succ', mul_assoc, mul_assoc, hx, hn]
```

Comment:
I would write (untested):
```suggestion
    (hx : ∀ y : R, f (x * y) = f x * f y) (n : ℕ) (y : R) : f (x ^ n * y) = f x ^ n * f y := by
  induction n generalizing y with
  | zero => rw [pow_zero, pow_zero, one_mul, one_mul]
  | succ n hn => rw [pow_succ', pow_succ', mul_assoc, mul_assoc, hx, hn]
```

Reply:
I tested your suggestion, but it does not work.